src/Tasks/Microsoft.NET.Build.Tasks/ProcessFrameworkReferences.cs

```diff
@@ -230,58 +229,74 @@ protected override void ExecuteCore()
                     isTrimmable = selectedRuntimePack?.IsTrimmable;
                 }
 
-                //  Only add runtime packs where the framework reference name matches the RuntimeFrameworkName
-                //  Framework references for "profiles" will use the runtime pack from the corresponding non-profile framework
+                bool includeInPackageDownload;
+                KnownRuntimePack runtimePackForRuntimeIDProcessing;
                 if (knownFrameworkReference.Name.Equals(knownFrameworkReference.RuntimeFrameworkName, StringComparison.OrdinalIgnoreCase))
                 {
-                    bool processedPrimaryRuntimeIdentifier = false;
+                    //  Only add runtime packs where the framework reference name matches the RuntimeFrameworkName
+                    //  Framework references for "profiles" will use the runtime pack from the corresponding non-profile framework
+                    runtimePackForRuntimeIDProcessing = selectedRuntimePack.Value;
+                    includeInPackageDownload = true;
+                }
+                else if (!knownFrameworkReference.RuntimePackRuntimeIdentifiers.Equals(selectedRuntimePack?.RuntimePackRuntimeIdentifiers))
+                {
+                    // If the profile has a different set of runtime identifiers than the runtime pack, use the profile.
+                    runtimePackForRuntimeIDProcessing = knownFrameworkReference.ToKnownRuntimePack();
+                    includeInPackageDownload = true;
+                }
+                else
```

Comment:
this "else" I am not too sure. Is it the same logic before https://github.com/dotnet/sdk/pull/11824 ?